calico_node/ip_detection.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+)
+
+func detectIPs(ignoreInterfaces []string) ([]string, error) {
+
+	ips := []string{}
+	ifaces, err := net.Interfaces()
+
+	for _, iface := range ifaces {
+
+		valid_ip := true
+
+		// Check each interface name against 
+		// list of interfaces to ignore.
+		for _, v := range ignoreInterfaces {
+			regex, err := regexp.Compile(v)
+
```

Comment:
I think generally it's good to keep the error checking close to the thing that's generating the error, so I'd remove this newline in between.